daft/expressions/expressions.py

```diff
@@ -1917,6 +1917,15 @@ def download(
         """
         multi_thread = ExpressionUrlNamespace._should_use_multithreading_tokio_runtime()
         io_config = ExpressionUrlNamespace._override_io_config_max_connections(max_connections, io_config)
+        if io_config.unity.endpoint is None:
+            from daft.catalog.__unity import UnityCatalog
+            from daft.session import current_catalog
+
+            catalog = current_catalog()
+            if isinstance(catalog, UnityCatalog):
+                unity_catalog = catalog._inner
+                io_config = io_config.replace(unity=unity_catalog.to_io_config().unity)
```

Comment:
Should we be doing this all the time? This causes an issue if someone did `pip install daft` but not `pip install daft[unity]`

Reply:
Good point. I'll look into lazy import of the unitycatalog library